tests/nosetests/pyanaconda_tests/module_scheduler_test.py

```diff
@@ -0,0 +1,417 @@
+            "error-messages": errors,
+            "warning-messages": warnings
+        })
+
+    def validate_mount_point_test(self):
+        """Test ValidateMountPoint."""
+        self._add_device(StorageDevice("dev1", fmt=get_format("ext4", mountpoint="/boot")))
+
+        report = self.interface.ValidateMountPoint("/boot")
+        self._check_report(report, "That mount point is already in use. Try something else?")
+
+        report = self.interface.ValidateMountPoint("")
+        self._check_report(report, "Please enter a valid mount point.")
+
+        report = self.interface.ValidateMountPoint("/dev")
+        self._check_report(report, "That mount point is invalid. Try something else?")
+
+        report = self.interface.ValidateMountPoint("/home/")
+        self._check_report(report, "That mount point is invalid. Try something else?")
+
+        report = self.interface.ValidateMountPoint("/home/")
+        self._check_report(report, "That mount point is invalid. Try something else?")
+
+        report = self.interface.ValidateMountPoint("home")
+        self._check_report(report, "That mount point is invalid. Try something else?")
+
+        report = self.interface.ValidateMountPoint("/ho me")
+        self._check_report(report, "That mount point is invalid. Try something else?")
+
+        report = self.interface.ValidateMountPoint("/home/../")
+        self._check_report(report, "That mount point is invalid. Try something else?")
+
+        report = self.interface.ValidateMountPoint("/home/..")
+        self._check_report(report, "That mount point is invalid. Try something else?")
+
+        report = self.interface.ValidateMountPoint("/home")
+        self._check_report(report, None)
+
+    def add_device_test(self):
+        """Test AddDevice."""
+        self._add_device(DiskDevice(
+            "dev1",
+            exists=True,
+            size=Size("15 GiB"),
+            fmt=get_format("disklabel")
+        ))
+
+        request = DeviceFactoryRequest()
+        request.device_type = DEVICE_TYPE_LVM
+        request.mount_point = "/home"
+        request.size = Size("5 GiB")
+        request.disks = ["dev1"]
+
+        self.storage.factory_device = Mock()
+        self.interface.AddDevice(get_native(DeviceFactoryRequest.to_structure(request)))
+        self.storage.factory_device.assert_called_once()
+
+    def validate_container_name_test(self):
+        """Test ValidateContainerName."""
+        report = self.interface.ValidateContainerName("_my/contain$er")
+        self._check_report(report, "Invalid container name. Try 'my_container'.")
+
+        report = self.interface.ValidateContainerName("my_container")
+        self._check_report(report, None)
+
+    def validate_raid_level_test(self):
+        """Test ValidateRaidLevel."""
+        report = self.interface.ValidateRaidLevel("raid6", 2)
+        self._check_report(report, "The RAID level you have selected raid6 requires more "
```

Comment:
Not actually related to this PR, but the "raid6" part is missing some quotes, brackets or the like, I think?

Reply:
Fixed.